frontend/src/components/project/Modify.js

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useEffect } from "react";
+import { Form, Input, Button, Mention } from "antd";
+
+import "codemirror/lib/codemirror.css";
+import "tui-editor/dist/tui-editor.min.css";
+import "tui-editor/dist/tui-editor-contents.min.css";
+import { Editor } from "@toast-ui/react-editor";
+
+import { getSampleProject } from "../../ProjectService";
+
+const ModifyProject = props => {
+    const [project, setProject] = useState({
+        id: null,
+        title: "loading...",
+        summary: "loading...",
+        content: "loading...",
+        tags: "software",
+        collaborators: ""
+    });
+    const allTags = ["software", "hardware", "math", "science", "laser"];
+    const allCollaborators = [
+        "bsmith@example.com",
+        "tom@example.com",
+        "johhny@rockets.net"
+    ];
+    const editorRef = React.createRef();
+    const { getFieldDecorator, setFieldsValue } = props.form;
+    const { toContentState } = Mention;
+
+    useEffect(() => {
+        let isSubscribed = true;
+        getSampleProject(props.match.params.id).then(
```

Comment:
I would honestly not call this `getSampleProject` but instead just `getProject` as if it is the real API. Then, whenever I am finished, you only need to change the API part and not all the code that is calling `getProject`.

This part of the codebase should not know that it is temporary API.